tests/Functional/FunctionalTest.php

```diff
@@ -94,12 +118,16 @@ public function getFunctionalTests(): array
         $tests = [];
 
         foreach ($finder as $dir) {
-            $rstFilename = $dir->getPathname() . '/' . $dir->getFilename() . '.rst';
-            if (! file_exists($rstFilename)) {
-                throw new Exception(sprintf('Could not find functional test file "%s"', $rstFilename));
+            $rstFilename   = $dir->getPathname() . '/' . $dir->getFilename() . '.rst';
+            $indexFilename = $dir->getPathname() . '/index.rst';
+            if (file_exists($rstFilename)) {
+                $rst = file_get_contents($rstFilename);
+            } elseif (file_exists($indexFilename)) {
+                $rst = file_get_contents($indexFilename) ?? '';
```

Comment:
```suggestion
                $rst = file_get_contents($indexFilename) ?? '';
```

I'm attempting to rebase this on 0.5.x and the newly updated phpstan tells me that the left expression is not nullable… let me try dropping the `?? ''` part, as I'm under the impression you fixed the previous issue some other way.

Reply:
Makes sense as we check weather the file exists